pkg/provisioner/ironic/ironic.go

```diff
@@ -82,36 +83,23 @@ func NewMacAddressConflictError(address, node string) error {
 	return macAddressConflictError{Address: address, ExistingNode: node}
 }
 
-func init() {
-	// NOTE(dhellmann): Use Fprintf() to report errors instead of
-	// logging, because logging is not configured yet in init().
-
-	var authErr error
-	ironicAuth, inspectorAuth, authErr = clients.LoadAuth()
-	if authErr != nil {
-		fmt.Fprintf(os.Stderr, "Cannot start: %s\n", authErr)
-		os.Exit(1)
-	}
-
+func loadConfigFromEnv() error {
 	deployKernelURL = os.Getenv("DEPLOY_KERNEL_URL")
-	if deployKernelURL == "" {
-		fmt.Fprintf(os.Stderr, "Cannot start: No DEPLOY_KERNEL_URL variable set\n")
-		os.Exit(1)
-	}
 	deployRamdiskURL = os.Getenv("DEPLOY_RAMDISK_URL")
-	if deployRamdiskURL == "" {
-		fmt.Fprintf(os.Stderr, "Cannot start: No DEPLOY_RAMDISK_URL variable set\n")
-		os.Exit(1)
+	deployISOURL = os.Getenv("DEPLOY_ISO_URL")
+	if deployISOURL == "" && (deployKernelURL == "" || deployRamdiskURL == "") {
+		return errors.New("Either DEPLOY_KERNEL_URL and DEPLOY_RAMDISK_URL or DEPLOY_ISO_URL must be set")
+	}
+	if (deployKernelURL == "" && deployRamdiskURL != "") || (deployKernelURL != "" && deployRamdiskURL == "") {
+		return errors.New("DEPLOY_KERNEL_URL and DEPLOY_RAMDISK_URL can only be set together")
```

Comment:
Ok, my personal inclination would be to specify exactly which expected vars were not set (without forcing the user to look at the current container configuration, especially useful when you have just the logs), but it's anyhow an improvement and don't consider it blocking